ui/src/components/WaveCanvas.js

```diff
@@ -70,16 +71,23 @@ export class WaveCanvas extends LitElement {
     updated(changedProperties) {
         if (
             this.audioStream != changedProperties.get('audioStream') &&
-            this.isRecording
+            this.getIsRecordingState()
         ) {
             this.soundWave.setStream(this.audioStream);
             this.soundWave.setContext(this.context);
             this.soundWave.createSoundWave();
-        } else if (!this.isRecording && this.soundWave != undefined) {
+        } else if (!this.getIsRecordingState() && this.soundWave != undefined) {
             this.soundWave.stopSoundWave();
         }
     }
 
+    /**
+     * @returns {Boolean} If the current local state property is the recording state
+     */
+    getIsRecordingState() {
```

Comment:
This property is used throughout, but it seems only the isRecording boolean is useful. Would it make more sense to pass down a boolean `isRecording` to these components? The other states aren't relevant.